library/core/helpers/class.ssoutils.php

```diff
@@ -36,14 +37,17 @@ class SsoUtils {
     /** @var string */
     private $stateToken;
 
+    private $logger;
+
     /**
      * SsoUtils constructor.
      */
-    public function __construct(Gdn_Configuration $config, Cookie $cookie, Gdn_Session $session) {
+    public function __construct(Gdn_Configuration $config, Cookie $cookie, Gdn_Session $session, Logger $logger) {
```

Comment:
This kinda thing has the potential to break backwards compatibility. You're adding a new required parameter to the `SsoUtils` constructor. If it's coming from the container, it's no big deal. But if anyone, anywhere, is manually instantiating this class, this change will break that. This constructor signature has been the same for about two years. It's very likely it's been manually instantiated somewhere in that time.

```suggestion
    public function __construct(Gdn_Configuration $config, Cookie $cookie, Gdn_Session $session, ?Logger $logger = null) {
```

You can check to see if `$logger` is null and grab it from the container, manually, if it is.

Reply:
I think you wanna switch up this type to `Psr\Log\LoggerInterface` too, to make this more flexible.